src/ape_accounts/accounts.py

```diff
@@ -119,11 +121,21 @@ def delete(self):
         self.__decrypt_keyfile(passphrase)
         self.keyfile_path.unlink()
 
-    def sign_message(self, msg: SignableMessage) -> Optional[MessageSignature]:
-        user_approves = self.__autosign or click.confirm(f"{msg}\n\nSign: ")
-        if not user_approves:
+    def sign_message(self, msg: Any, **signer_options) -> Optional[MessageSignature]:
+        if not isinstance(msg, (SignableMessage, EIP712Message, str)):
+            logger.warning("Unsupported message type, (type=%r, msg=%r)", type(msg), msg)
             return None
 
+        user_approves = self.__autosign or click.confirm(f"Message: {msg}\n\nSign: ")
+        if isinstance(msg, str):
+            # Convert str to SignableMessage for handling below
+            msg = encode_defunct(text=msg)
+        elif isinstance(msg, EIP712Message):
```

Comment:
future design idea: I wonder if we should have our conversion system also supply hooks for signing messages, like we do for tx but for messages.

then we could literally build an eip712 plugin and wouldnt need to add these types every in core.

for example, what if starknet or something offers a whole new spec for message signing, we could also implement conversion hooks there...